packages/ketcher-core/src/application/render/renderers/PolymerBondRenderer/SnakeModeHydrogenBondRenderer.ts

```diff
@@ -0,0 +1,962 @@
+import { SnakeMode } from 'application/editor';
+import { editorEvents } from 'application/editor/editorEvents';
+import { CoreEditor } from 'application/editor/internal';
+import { Coordinates } from 'application/editor/shared/coordinates';
+import type { PolymerBondRendererStartAndEndPositions } from 'application/render/renderers/PolymerBondRenderer/PolymerBondRenderer.types';
+import { SideChainConnectionBondRendererUtil } from 'application/render/renderers/PolymerBondRenderer/SideChainConnectionBondRendererUtil';
+import { SVGPathDAttributeUtil } from 'application/render/renderers/PolymerBondRenderer/SVGPathDAttributeUtil';
+import { D3SvgElementSelection } from 'application/render/types';
+import assert from 'assert';
+import { Vec2 } from 'domain/entities';
+import { Cell } from 'domain/entities/canvas-matrix/Cell';
+import {
+  Connection,
+  ConnectionXDirectionInDegrees,
+} from 'domain/entities/canvas-matrix/Connection';
+import { DrawingEntity } from 'domain/entities/DrawingEntity';
+import { PolymerBond } from 'domain/entities/PolymerBond';
+import { getSugarFromRnaBase } from 'domain/helpers/monomers';
+import { BaseRenderer } from '../BaseRenderer';
+import {
+  CORNER_LENGTH,
+  DOUBLE_CORNER_LENGTH,
+  generateCornerFromBottomToRight,
+  generateCornerFromLeftToBottom,
+  generateCornerFromLeftToTop,
+  generateCornerFromRightToBottom,
+  generateCornerFromRightToTop,
+  generateCornerFromTopToLeft,
+  generateCornerFromTopToRight,
+} from './helpers';
+
+enum LineDirection {
+  Horizontal = 'Horizontal',
+  Vertical = 'Vertical',
+}
+
+const LINE_FROM_MONOMER_LENGTH = 15;
+const VERTICAL_LINE_LENGTH = 21;
+const RNA_CHAIN_VERTICAL_LINE_LENGTH = 74;
+const RNA_ANTISENSE_CHAIN_VERTICAL_LINE_LENGTH = 20;
+const RNA_SENSE_CHAIN_VERTICAL_LINE_LENGTH = 210;
+
+const BOND_END_LENGTH = 15;
+const SIDE_CONNECTION_BODY_ELEMENT_CLASS = 'polymer-bond-body';
+
+export class SnakeModeHydrogenBondRenderer extends BaseRenderer {
```

Comment:
Render part of hydrogen bond is almost the same as for side connection bond. There is no need to separate them. >95% of code is the same as for side connection